src/GeniusYield/TxBuilder/Run.hs

```diff
@@ -187,11 +181,10 @@ instance GYTxMonad GYTxMonadRun where
 
     someUTxO = do
         addr        <- ownAddress
-        mcollateral <- asks runEnvCollateral
         utxos       <- utxosAtAddress addr
-        case find (\ref -> mcollateral /= Just ref) $ utxosRefs utxos of
+        case uncons $ utxosRefs utxos of
             Nothing  -> throwError $ GYQueryUTxOException $ GYNoUtxosAtAddress [addr]
-            Just ref -> return ref
+            Just (ref, _) -> return ref
```

Comment:
I think it will be better to just use list PM
```
case utxosRefs utxos of
            []    -> ...
            ref:_ ->  ...
```

Reply:
Yes :+1: